index.bs

```diff
@@ -9024,6 +9070,386 @@ worker=] algorithm:
+
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+
+1. Return |cookies|.
+
+</div>
+
+### Commands ### {#module-storage-commands}
+
+#### The storage.getCookies Command ####  {#command-storage-getCookies}
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = (
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        )
+
+        <!--
+        Modifications to this definition should be reflected in
+        `network.Cookie`, `network.SetCookieHeader`, and `storage.PartialCookie`.
+        -->
+        storage.CookieFilter = {
+          ? name: text,
+          ? value: network.BytesValue,
+          ? domain: text,
+          ? path: text,
+          ? size: js-uint,
+          ? httpOnly: bool,
+          ? secure: bool,
+          ? sameSite: network.SameSite,
+          ? expiry: js-uint,
+          Extensible,
+        }
+
+        storage.BrowsingContextPartitionDescriptor = {
+          type: "context",
+          context: browsingContext.BrowsingContext
+        }
+
+        storage.StorageKeyPartitionDescriptor = {
+          type: "storageKey",
+          ? userContext: text,
+          ? sourceOrigin: text,
+          Extensible,
+        }
+
+        storage.PartitionDescriptor = (
+          storage.BrowsingContextPartitionDescriptor /
+          storage.StorageKeyPartitionDescriptor
+        )
+
+        storage.GetCookiesParameters = {
+          ? filter: storage.CookieFilter,
+          ? partition: storage.PartitionDescriptor,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.GetCookiesResult = {
+        cookies: [*network.Cookie],
+        partitionKey: storage.PartitionKey,
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for storage.getCookies">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |filter| be the value of the <code>filter</code> field of |command parameters|
+  if it is present or an empty [=/map=] if it isn't.
+
+1. Let |partition spec| be the value of the <code>partition</code> field of
+  |command parameters| if it is present or null if it isn't.
+
+1. Let |partition key| be the result of [=trying=] to [=expand a storage partition spec|expand=] |partition spec|.
+
+1. Let |store| be the result of [=trying=] to [=get the cookie store=] with
+  |partition key|.
+
+1. Let |cookies| be the result of [=get matching cookies=] with |store| and |filter|.
+
+1. Let |body| be a [=/map=] matching the <code>storage.GetCookiesResult</code> production,
+  with the <code>cookies</code> field set to |cookies| and the <code>partitionKey</code>
+  field set to |partition key|.
```

Comment:
At this point the partition key can be an empty map (if `partition spec` is was set, and endpoint does not require `userContext` and `sourceOrigin`), which conflicts with the result type `partitionKey: storage.PartitionKey`.

Reply:
Addressed via commit 35d7fad.